@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -286,8 +307,121 @@ export function placeIcon(
+): PlacementResult {
+    const label = labelState.element;
+
+    // NOTE:
+    // We may save some CPU time by firstly checking visibility of all possible text placements,
+    // their bounding boxes merged altogether, this may save time for further calculations.
+    // Would be enough to include TL and BR placements.
+
+    // Start with currently set alignment settings.
+    const basePlacement = {
+        h: label.layoutStyle!.horizontalAlignment,
+        v: label.layoutStyle!.verticalAlignment
+    };
+    // Placements options will be read from label.layoutStyle.placements in final solution.
+    const placements = possibleAnchorPlacementsCW;
+    // Find first anchor placement.
+    const idx = placements.findIndex(val => {
+        return val.h === basePlacement.h && val.v === basePlacement.v;
+    });
+    // If not listed in placement options list (idx === -1) will start from current alignment setup.
+    const placementsNum = placements.length;
+    // Will be true if all text placements are invisible.
+    let allInvisible: boolean = true;
+    let i = 0;
+    for (i = 0; i < placementsNum; ++i) {
+        const anchorPlacement = idx + i < 0 ? basePlacement : placements[(idx + i) % placementsNum];
+        // Override label text alignment for measurements and leave it so if passed collisions test.
+        // NOTE: This actually writes to label.layoutStyle.verticalAlignment/horizontalAlignment.
+        textCanvas.textLayoutStyle.horizontalAlignment = anchorPlacement.h;
+        textCanvas.textLayoutStyle.verticalAlignment = anchorPlacement.v;
+
+        // Compute label bounds, visibility or collision according to new layout settings.
+        const placementResult = placePointLabel(
+            labelState,
+            screenPosition,
+            scale,
+            textCanvas,
+            screenCollisions,
+            isRejected,
+            true,
+            outScreenPosition
+        );
+
+        // Check the text visibility.
+        if (placementResult === PlacementResult.Invisible) {
+            continue;
+            // This placement if visible, check for collisions.
+        } else {
+            allInvisible = false;
+        }
+
+        // If icon rejected proceed to test further if all elements are invisible.
+        // TODO:
+        // If icon's is isRejected, should we test visibility of all placements, maybe better to
+        // interrupt tests now and save some CPU time?
+        if (placementResult === PlacementResult.Rejected && isRejected) {
+            continue;
+        }
+        // Check label's text collision.
+        // TODO: Same comment as above, maybe better to interrupt the tests here after checking
+        // first placement and save CPU time.
+        if (placementResult === PlacementResult.Rejected) {
+            continue;
+        }
+
+        // Glyphs arrangement have been changed remove text buffer object.
+        if (i !== 0) {
+            delete label.textBufferObject;
```

Comment:
Afaik this delete is not needed, setting the bounds to undefined will suffice. I haven't seen this idiom used in our codebase, at least applied to typescript class members.

Reply:
Done.